### DISCUSSION TITLE
Unit status

### DISCUSSION THREAD

**Discussion by federicopeyrani1969** [orbat-mapper/orbat-mapper#141] [Q&A] [answered]
Do you plan to add a "Status" to an unit?
I mean Status(es) could be: forming, training, on the move, attacking, defending, recovering, disbanding, converting to another unit, and so on. I am curious!
Federico

**Comment by kjellmf** [comment]
Good idea. We already kind of have this through the status field in the symbol identification code, but the values are limited to a few values (present/damaged/destroyed..). A more general status attribute will be a nice addition. You mention some potential values. Are you aware of others? Maybe there is some kind of standard available?

**Reply by kjellmf** [reply]
Added issue #142 to keep track of this.

**Reply by federicopeyrani1969** [reply]
Hi Kjellmf
I personally don't know if there's a standard for the statuses, my list would be:
1) Forming 2) Training 3) Ready - Stationary/Resting 4) Ready - In movement 5) In combat - Manning a Line 6) In combat - Advancing 7) In combat - Retreating 8) Disbanding 9) Merging with Unit XYZ.

Maybe you could also add a force indicator (like a pecentage bar), indicating if the unit was full effective or depleted. 
German units reported this monthly for men and equipment of every subunit!

Regards
Federico

**Reply by federicopeyrani1969** [reply]
Just come to my mind a status: 10) Remains of the unit
Ciao
Federico

**Reply by kjellmf** [reply]
I found this [US ARMY UNIT STATUS REPORTING AND FORCE REGISTRATION-CONSOLIDATED POLICIES](https://armypubs.army.mil/ProductMaps/PubForm/Details.aspx?PUB_ID=1021857).

**Reply by federicopeyrani1969** [reply]
This is very articulated! 
I seem to remember that Germans in WWII only had three statuses: 
fit for every type of Mission, fit for defense, and fit only for very limited tasks.
Federico

**Comment by kjellmf** [comment, marked as answer]
I've now added basic support for unit status. The user interface is a bit awkward, but it is a start. The different status values are user defined. You can see some example values in the Narvik scenario https://orbat-mapper.app/scenario/demo-narvik40

**Reply by federicopeyrani1969** [reply]
Have seen the Narvik demo, looks great! I ask myself if unit status has effects to unit icon, so that one can see it visually without opening the menu. Maybe STANAG has some indication on how to show different statuses.

**Reply by kjellmf** [reply]
APP6/2525 has unit status as part of the symbol identification code, but it is limited to the following values:

```
{ code: "0", text: "Present" },
{ code: "1", text: "Planned/Anticipated/Suspect" },
{ code: "2", text: "Present/Fully capable" },
{ code: "3", text: "Present/Damaged" },
{ code: "4", text: "Present/Destroyed" },
{ code: "5", text: "Present/Full to capacity" },
```

I could probably use a text modifier to show the unit status on the map.

**Reply by federicopeyrani1969** [reply]
Which is the difference between code 2 and code 5? It's not clear to me...
Interesting the "Planned/Anticipated/Suspected"

In the end I think that your choice is better, specifically: attacking defending moving. They explain better what to expect form the unit on the map.

**Reply by kjellmf** [reply]
From the APP-6E document: "Full to Capacity" is associated with a symbolized object where its capacity can be measured, and the status of that capacity is relevant
